pkg/codegen/properties.go

```diff
@@ -194,6 +194,23 @@ func getType(name string, schema *schemas.JSONSchema, required *bool, typePrefix
 			for k := range schema.Properties {
 				typeStr = strcase.ToCamel(k)
 			}
+		case schema.AdditionalProperties.JSONSchema != nil:
+			if len(schema.Properties) > 0 {
+				return "", fmt.Errorf("cannot determine the type of object %s: properties and additionalProperties are both defined", name)
+			}
+			itemType, err := getType(name, schema.AdditionalProperties.JSONSchema, nil, "", false)
```

Comment:
Yes good point! I've also been unable to find any explicit information regarding this in the JSON schema specification from multiple resources, including:

- [Understanding JSON Schema](https://json-schema.org/understanding-json-schema/reference/object#additionalproperties)
- [RFC 8927](https://www.rfc-editor.org/rfc/rfc8927.html#name-allowing-additional-propert)

Interestingly, these two sources seem to contradict each other on the default behavior when `additionalProperties` is not set. One suggests that additional properties are allowed by default, while the other implies the opposite.

To gain a better understanding, I conducted some tests using the [online JSON schema editor](https://app.quicktype.io/?l=go). I experimented with placing an object definition in `additionalProperties` and found that it does indeed generate the sub-object as expected. Here's the JSON schema I used:

```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "QueryMsg",
  "description": "Query messages",
  "oneOf": [
    {
      "title": "MapString",
      "type": "object",
      "required": ["map_string"],
      "properties": {
        "map_string": {
          "type": "object",
          "required": ["foo"],
          "properties": {
            "foo": {
              "type": "object",
              "additionalProperties": {
                "type": "object",
                "properties": {
                  "a": {"type": "string"}
                }
              }
            }
          },
          "additionalProperties": false
        }
      },
      "additionalProperties": false
    }
  ]
}
```

And here's the corresponding Go code that was generated:

```go
// Query messages
type Query struct {
  MapString MapString `json:"map_string"`
}

type MapString struct {
  Foo map[string]Foo `json:"foo"`
}

type Foo struct {
  A *string `json:"a,omitempty"`
}
```

While this test indicates that such a JSON schema is possible, I haven't come across any instances of this in the JSON schemas generated from smart contracts and Rust code. Maybe adding an assertion on this can be a good starting point.

Reply:
This Go code wasn't generated by go-codegen right? And yes, let's add the assertion and merge this then

Reply:
The code in question was generated using [quicktype](https://quicktype.io), specifically through their [online editor](https://app.quicktype.io). 

Thanks, in response to your feedback, I've incorporated an assertion to verify if `additionalProperties` contains properties. If this is the case, the code will now return an error. I believe this check should be sufficient and appropriate for identifying sub-definitions.